File: org.openhab.binding.vera/src/main/java/org/openhab/binding/vera/luup/Scene.java
package org.openhab.binding.vera.luup;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Scene {
    private int id;

    private String name;

    @JsonProperty("room")
    private int roomId;

    private int state;

    private boolean active;

    private String comment;

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public int getRoomId() {
        return roomId;
    }

    public int getState() {
        return state;
    }

    public boolean isActive() {
        return active;
    }

    public String getComment() {
        return comment;
    }
}

File: org.openhab.binding.vera/src/main/java/org/openhab/binding/vera/luup/Scene.scala
package org.openhab.binding.vera.luup;

import scala.beans.BeanProperty

import com.fasterxml.jackson.annotation.JsonProperty;

class Scene {
    @BeanProperty
    protected var id: Int = _

    @BeanProperty
    protected var name: String = _

    @JsonProperty("room")
    @BeanProperty
    protected var roomId: Int = _

    @BeanProperty
    protected var state: Int = _

    @BeanProperty
    protected var active: Boolean = _

    @BeanProperty
    protected var comment: String = _

    override def toString(): String = s"Scene[$id]: $name"
}
